```
{
  "CVE-2018-10855": {
    "description": "ansible: Failed tasks do not honour no_log option allowing for secrets to be disclosed in logs",
    "root_cause": "The `no_log` option in Ansible is not properly honored when a task fails due to some exceptions from connection plugins or during iterations.",
    "vulnerabilities": [
      "Information Leakage",
      "Sensitive data exposure in logs",
      "Failure to honor no_log task flag"
    ],
    "impact": "Sensitive information such as secrets passed to a task could be disclosed in logs or on the terminal when a task fails, despite the use of the no_log flag.",
    "attack_vectors": [
      "By failing an Ansible task with sensitive data",
         "By triggering a connection exception",
         "Having a task with no_log option fail during iterations"
    ],
    "attacker_capabilities": "An attacker needs the ability to trigger a task failure to exploit this vulnerability. This can be done by controlling parameters passed to the task that cause a failure or by inducing connection errors",
  "references": [
        "https://access.redhat.com/errata/RHSA-2018:2079",
        "https://bugzilla.redhat.com/show_bug.cgi?id=1588855",
         "https://access.redhat.com/errata/RHSA-2018:1948",
         "https://access.redhat.com/errata/RHSA-2018:1949",
         "https://access.redhat.com/errata/RHSA-2018:2022",
          "https://access.redhat.com/errata/RHSA-2018:2184",
         "https://access.redhat.com/errata/RHSA-2018:2585",
        "https://access.redhat.com/errata/RHSA-2019:0054",
        "https://www.debian.org/security/2019/dsa-4396",
          "https://access.redhat.com/errata/RHSA-2018:2321",
       "https://access.redhat.com/security/cve/CVE-2018-10855.html",
        "https://access.redhat.com/errata/RHBA-2018:3788"
       ]
  }
}
```